lib/modem_info/modem_info_json.c

```diff
@@ -179,67 +179,52 @@ static int device_data_add(struct device_param *device, cJSON *json_obj)
 int modem_info_json_object_encode(struct modem_param_info *modem,
 				  cJSON *root_obj)
 {
-	int total_len = 0;
-	int ret;
-
 	if (root_obj == NULL || modem == NULL) {
 		return -EINVAL;
 	}
 
+	int obj_count = cJSON_GetArraySize(root_obj);
+
 	cJSON *network_obj	= cJSON_CreateObject();
 	cJSON *sim_obj		= cJSON_CreateObject();
 	cJSON *device_obj	= cJSON_CreateObject();
 
 	if (network_obj == NULL || sim_obj == NULL || device_obj == NULL) {
-		total_len = -ENOMEM;
+		obj_count = -ENOMEM;
 		goto delete_object;
 	}
 
-	if (IS_ENABLED(CONFIG_MODEM_INFO_ADD_NETWORK)) {
-		ret = network_data_add(&modem->network, network_obj);
+	if (IS_ENABLED(CONFIG_MODEM_INFO_ADD_NETWORK) &&
+	    network_data_add(&modem->network, network_obj)) {
```

Comment:
This check is no good - `network_data_add` might return a negative error code and this will slip in undetected.